scipy/signal/_polyutils.py

```diff
@@ -0,0 +1,83 @@
+"""Partial replacements for numpy polynomial routines, with Array API compatibility.
+
+This module contains both "old-style", np.poly1d, routines from the main numpy
+namespace, and "new-style", np.polynomial.polynomial, routines.
+
+To distinguish the two sets, the "new-style" routine names start with `npp_`
+"""
+import scipy._lib.array_api_extra as xpx
+
+
+def polyroots(coef, *, xp):
+    """numpy.roots, best-effor replacement
```

Comment:
This routine can potentially be considered for `array-api-extra`. That said, it might not be worth it:
- it's basically the `# companion matrix` part below
- the real blocker is `eigvals`: it is not standardized, and it was previously not available on CUDA at all. Now the low-level blocker is removed (https://github.com/cupy/cupy/issues/6359#issuecomment-2425994358), so CuPy is hopefully getting `eig / eigvals` soon (https://github.com/cupy/cupy/pull/8980). If JAX CUDA grows this routine, too, and we can have it in the array API spec, then SciPy can just retain the companion matrix computation.